### PR TITLE
Fix match clause for non-string values

### PR DESCRIPTION
**Pull request by AsafMah**

### CODE REVIEW

File: K2Bridge/JsonConverters/MatchPhraseClauseConverter.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ internal class MatchPhraseClauseConverter : ReadOnlyJsonConverter
                 var obj = new MatchPhraseClause
                 {
                     FieldName = first.Name,
-                    Phrase = (string)((JValue)first.First).Value,
+                    Phrase = ((JValue)first.First).Value?.ToString() ?? string.Empty,
```
Comment (eladiw) [inline K2Bridge/JsonConverters/MatchPhraseClauseConverter.cs:42]: can you add unit tests for the two options?

**Review by YannickBrombach — APPROVED** [review]
Looking good to me!

**Review by eladiw — CHANGES_REQUESTED** [review]
please add UTs

File: K2Bridge/ElasticQueryTranslator.cs [hunk 2]
```diff
@@ -83,7 +83,7 @@ public QueryData TranslateQuery(string header, string query)
                                 elasticSearchDsl.HighlightText.Add("*", queryStringClause.Phrase);
                                 break;
                             case MatchPhraseClause matchPhraseClause:
-                                elasticSearchDsl.HighlightText.Add(matchPhraseClause.FieldName, matchPhraseClause.Phrase);
+                                elasticSearchDsl.HighlightText.Add(matchPhraseClause.FieldName, matchPhraseClause.Phrase.ToString()); // TODO - fix this when highlighting is fixed
```
Comment (eladiw) [inline K2Bridge/ElasticQueryTranslator.cs:86]: what is this todo? I am not aware of any issue with the highlighting

File: K2Bridge.Tests.UnitTests/Visitors/MatchPhraseVisitorTests.cs [hunk 3]
```diff
@@ -19,6 +20,30 @@ public string MatchPhraseVisit_WithValidClause_ReturnsEquals()
...
 
+        [TestCase(ExpectedResult = "MyField == \"3\"")]
+        public string MatchPhraseVisit_WithValidIntClause_ReturnsEquals()
+        {
+            var matchPhraseClause = CreateMatchPhraseClause("MyField", 3);
+
+            return VisitQuery(matchPhraseClause);
+        }
+
+        [TestCase(ExpectedResult = "MyField == \"3.5\"")]
```
Comment (eladiw) [inline K2Bridge.Tests.UnitTests/Visitors/MatchPhraseVisitorTests.cs:31]: is this correct? should 3.5 treated as string?

**Review by eladiw — CHANGES_REQUESTED** [review]
some questions

File: K2Bridge/ElasticQueryTranslator.cs (hunk 2 above)
Comment (AsafMah) [reply K2Bridge/ElasticQueryTranslator.cs:86]: Do you know how can I actually get to this line?

When adding a match clause in the UI it's always under elasticSearchDsl.Query.Bool.Filter

File: K2Bridge.Tests.UnitTests/Visitors/MatchPhraseVisitorTests.cs (hunk 3 above)
Comment (AsafMah) [reply K2Bridge.Tests.UnitTests/Visitors/MatchPhraseVisitorTests.cs]: It does work, but now I changed it to be more idiomatic

File: K2Bridge/ElasticQueryTranslator.cs (hunk 2 above)
Comment (eladiw) [reply K2Bridge/ElasticQueryTranslator.cs:86]: try the filters

**Review by eladiw — APPROVED** [review]

**Pull request merged**